balancer-js/src/modules/data/pool/subgraph.ts

```diff
@@ -33,20 +39,30 @@ export class PoolsSubgraphRepository
     this.client = createSubgraphClient(url);
   }
 
-  async fetch(): Promise<SubgraphPool[]> {
-    const { pool0, pool1000 } = await this.client.Pools({
-      where: { swapEnabled: true, totalShares_gt: '0' },
+  async fetch(query?: GraphQLQuery): Promise<Pool[]> {
+    const defaultArgs: GraphQLArgs = {
       orderBy: Pool_OrderBy.TotalLiquidity,
       orderDirection: OrderDirection.Desc,
       block: this.blockHeight
         ? { number: await this.blockHeight() }
         : undefined,
-    });
+      where: {
+        swapEnabled: Op.Equals(true),
+        totalShares: Op.GreaterThan(0),
```

Comment:
The reason we need this is because arguments for Subgraph look like:
```
where: {
    swapEnabled: true,
    totalShares_gt: 0
}
```
While the arguments for balancer API look like:
```
where: {
    swapEnabled: {
        eq: true
    },
    totalShares: {
        gt: 0
    }
}
```

And we need a way for users to specify the arguments once and have them work with both. How else would you do it?

Reply:
now i get it, it's to standardise subgraph and amazon graphql queries?

Reply:
Yep. I'm thinking removing the `Op` now and just passing in the where query in AWS style and the Subgraph can reformat it manually.